src/Dom/Document/Filter/DeduplicateTag.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace AmpProject\Dom\Document\Filter;
+
+use AmpProject\Dom\Document;
+use AmpProject\Dom\Document\AfterLoadFilter;
+use AmpProject\Tag;
+
+/**
+ * Filter for deduplicating head and body tags.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class DeduplicateTag implements AfterLoadFilter
+{
+
+    /**
+     * Deduplicate head and body tags.
+     *
+     * This keeps the first tag as the main tag and moves over all child nodes and attribute nodes from any subsequent
+     * same tags over to remove them.
+     *
+     * @param Document $document Document to be processed.
+     */
+    public function afterLoad(Document $document)
+    {
+        $tagNames = [Tag::HEAD, Tag::BODY];
+
+        foreach ($tagNames as $tagName) {
+            $tags = $document->getElementsByTagName($tagName);
+
+            /**
+             * Main tag to keep.
+             *
+             * @var \AmpProject\Dom\Element|null $mainTag
```

Comment:
Import should be used here:
```suggestion
             * @var Element|null $mainTag
```

Reply:
Updated the code.